File: Interviews/GFG/Daily-Problems/2022/October/Day-30/Chicks-in-a-Zoo.cpp
class Solution {
public:
	long long int NoOfChicks (int n) {
		vector<long long int> dp(n, 1);
		long long int ans = 1;
		for (int i = 1; i < n; i++) {
			if (i >= 6) {
				ans -= dp[i - 6];
			}
			dp[i] = ans * 2;
			ans += ans * 2;
		}

		return ans;
	}
};

File: Interviews/GFG/random-problems/Easy/Chicks-in-a-Zoo/Chicks-in-a-Zoo.py
class Solution:
	def NoOfChicks (self, N):
		dp, ans = [1 for i in range(N)], 1
		for i in range(1, N):
		    if i >= 6: ans -= dp[i - 6]
		    dp[i] = ans * 2
		    ans += ans * 2
		
		return ans